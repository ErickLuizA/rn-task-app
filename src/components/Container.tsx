import * as React from 'react'
import { StyleSheet, StatusBar } from 'react-native'
import { useTheme } from 'react-native-paper'
import { SafeAreaView } from 'react-native-safe-area-context'

interface IProps {
  upper?: boolean
}

const styles = StyleSheet.create({
  container: {
    paddingTop: StatusBar.currentHeight,
    paddingHorizontal: 20,
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },

  smallContainer: {
    paddingTop: StatusBar.currentHeight,
    paddingHorizontal: 20,
    flex: 1,
    alignItems: 'center',
  },
})

const Container: React.FC<IProps> = ({ children, upper }) => {
  const { colors } = useTheme()

  return (
    <SafeAreaView
      testID="container"
      style={[
        { backgroundColor: colors.background },
        upper ? styles.smallContainer : styles.container,
      ]}>
      {children}
    </SafeAreaView>
  )
}

export default Container
