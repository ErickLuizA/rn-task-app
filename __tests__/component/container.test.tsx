import * as React from 'react'
import { render } from 'react-native-testing-library'
import Container from '../../src/components/Container'

describe('<Container />', () => {
  it('render the container component', () => {
    const tree = render(<Container />).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
