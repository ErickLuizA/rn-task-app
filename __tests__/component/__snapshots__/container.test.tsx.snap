// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container /> render the container component 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "backgroundColor": "#f6f6f6",
      },
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
        "paddingHorizontal": 20,
        "paddingTop": null,
      },
    ]
  }
  testID="container"
/>
`;
